js/interface.js

```diff
@@ -560,21 +560,26 @@ $('#app')
   })
   .on('click', '[data-create-source]', function(event) {
     event.preventDefault();
-    var sourceName = prompt('Please type the new table name:');
-
-    if (!sourceName) {
-      return;
-    }
+    Fliplet.Modal.prompt({
+      title: 'New data source',
+      message: 'Enter a data source name',
+    }).then(function (result) {
+      var dataSourceName = result.trim();
+      if (!dataSourceName) {
```

Comment:
The result could come back with `null` (cancelled) or `""`. They should be treated differently.

If it's cancelled, no further action should be taken, including any alerts.

If it comes back with an empty string, the alert defined here should be shown. After the alert is dismissed, the user should be shown the same prompt modal again so they can continue.